src/test/java/org/openrewrite/java/migrate/lang/var/UseVarForGenericsConstructorsTest.java

```diff
@@ -143,6 +146,179 @@ void m() {
+            );
+        }
+
+        @Test
+        @Disabled("AST look wrong for OpenRewrite but its valid")
+        void unboundedGenerics(){
+            //language=java
+            rewriteRun(
+              version(
+                java("""
+                  package com.example.app;
+                              
+                  import java.util.List;
+                  import java.util.ArrayList;
+                                    
+                  class A {
+                      void generic() {
+                          List<?> lst = new ArrayList<>();
+                      }
+                  }
+                  ""","""
+                  package com.example.app;
+                              
+                  import java.util.List;
+                  import java.util.ArrayList;
+                                    
+                  class A {
+                      void generic() {
+                          var lst = new ArrayList<?>();
+                      }
+                  }
+                  """),
+                10
+              )
+            );
+        }
+
+            @Test
+            @ExpectedToFail("Fails with IllegalStateException because not yet implemented")
+            void boundedGenerics(){
+                //language=java
+                rewriteRun(
+                  version(
+                    java("""
+                  package com.example.app;
+                              
+                  import java.util.List;
+                  import java.util.ArrayList;
+                                    
+                  class A {
+                      void generic() {
+                          List<? extends String> lst = new ArrayList<>();
+                      }
+                  }
+                  ""","""
+                  package com.example.app;
+                              
+                  import java.util.List;
+                  import java.util.ArrayList;
+                                    
+                  class A {
+                      void generic() {
+                          var lst = new ArrayList<? extends String>();
+                      }
+                  }
+                  """),
+                    10
+                  )
+                );
+            }
+
+        @Test
+        void inceptionGenerics() {
+            //language=java
+            rewriteRun(
+              version(
+                java("""
+                  package com.example.app;
+                              
+                  import java.util.List;
+                  import java.util.ArrayList;
+                                    
+                  class A {
+                      void generic() {
+                          List<List<Object>> lst = new ArrayList<>();
+                      }
+                  }
+                  """, """
+                  package com.example.app;
+                              
+                  import java.util.List;
```

Comment:
Hmm; we might want to clear out unused imports.
```suggestion
```

Reply:
Thought about that too but added a note here [java-lang-var.yml L20](https://github.com/openrewrite/rewrite-migrate-java/blob/main/src/main/resources/META-INF/rewrite/java-lang-var.yml#L20)

Reply:
Hmm; but wouldn't we at least be able to `maybeRemoveImport` the type on left side of the assignment when replaced with `var` (or maybe only if the left side is an interface)? If there's any other uses still then the `maybe` part should pick up on that.

Reply:
Oh wasn't aware of this. This should be rolled out to every Recipe in `org.openrewrite.java.migrate.lang.var`